arm-containerregistry/2017-03-01/swagger/containerregistry.json

```diff
@@ -0,0 +1,613 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/RegenerateCredentialParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the specified credential regenerated successfully.",
+            "schema": {
+              "$ref": "#/definitions/RegistryListCredentialsResult"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "RegistryNameCheckRequest": {
+      "description": "A request to check whether the container registry name is available.",
+      "required": [
+        "name",
+        "type"
+      ],
+      "properties": {
+        "name": {
+          "description": "The name of the container registry.",
+          "type": "string"
+        },
+        "type": {
+          "description": "The resource type of the container registry. This field must be set to \"Microsoft.ContainerRegistry/registries\".",
+          "enum": [
+            "Microsoft.ContainerRegistry/registries"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ContainerRegistryResourceType",
+            "modelAsString": false
+          }
+        }
+      }
+    },
+    "RegistryNameStatus": {
+      "description": "The result of a request to check the availability of a container registry name.",
+      "properties": {
+        "nameAvailable": {
+          "description": "The value that indicates whether the name is available.",
+          "type": "boolean"
+        },
+        "reason": {
+          "description": "If any, the reason that the name is not available.",
+          "type": "string"
+        },
+        "message": {
+          "description": "If any, the error message that provides more detail for the reason that the name is not available.",
+          "type": "string"
+        }
+      }
+    },
+    "Registry": {
+      "description": "An object that represents a container registry.",
+      "required": [
+        "sku"
+      ],
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "properties": {
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The Sku of the container registry."
+        },
+        "properties": {
+          "$ref": "#/definitions/RegistryProperties",
+          "description": "The properties of the container registry.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "Sku": {
+      "description": "The Sku of the container registry.",
+      "properties": {
+        "name": {
+          "description": "Name of the registry Sku. Required for registry creation.",
+          "enum": [
+            "Basic"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "SkuName",
+            "modelAsString": false
```

Comment:
We would update the swagger if we want to add/update the enum values. It may be better to be consistent between the spec and RP rather than accept an enum value that is not in spec, since we already have the enum values in the swagger. Is this an acceptable workaround?